src/gracenote.ts

```diff
@@ -1,24 +1,35 @@
 // [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
 
-import { StaveNote } from './stavenote';
+import { Vex } from './vex';
+import { StaveNote, StaveNoteStruct } from './stavenote';
 import { Stem } from './stem';
 import { Flow } from './tables';
 
+export interface GraceNoteStruct extends StaveNoteStruct {
+  slash: boolean;
+}
 export class GraceNote extends StaveNote {
-  static get CATEGORY() {
+  protected slash: boolean;
+  protected slur: boolean;
+
+  static get CATEGORY(): string {
     return 'gracenotes';
   }
-  static get LEDGER_LINE_OFFSET() {
+
+  static get LEDGER_LINE_OFFSET(): number {
     return 2;
   }
-  static get SCALE() {
+
+  static get SCALE(): number {
     return 0.66;
   }
 
-  constructor(note_struct) {
+  constructor(note_struct: GraceNoteStruct) {
     super({
-      glyph_font_scale: Flow.DEFAULT_NOTATION_FONT_SCALE * GraceNote.SCALE,
-      stroke_px: GraceNote.LEDGER_LINE_OFFSET,
+      ...{
```

Comment:
This looks strange, what's wrong with:

```typescript
super({
   glyph_font_scale: Flow.DEF...,
   stroke_px: GraceNote.LE...,
   ...note_struct,
});

Reply:
The compiler complains because the attributes are not optional. The compilers thinks that this makes no sense. You explained that we still have to do it due to javascript. If you use two spread operators the compiler does not complain anymore.

Reply:
@0xfe is that fine?

Reply:
Ah, okay, that's good.